src/main/java/cn/nukkit/player/Player.java

```diff
@@ -1554,6 +1555,12 @@ public boolean onUpdate(int currentTick) {
             this.updateData();
         }
 
+        setFlag(
+                EntityFlag.BLOCKING,
+                this.isSneaking() && this.getInventory().getItemInHand().getId() == ItemIds.SHIELD // Add support to off hand
+        );
+        updateData();
+
```

Comment:
Thank you for the PR! I think this should be inside the timing `try` block, right before the `this.updateData();` statement above, this way the timing would be tracked and we wouldn't need to update the data twice.

Reply:
I meat this `this.updateData();` call:
https://github.com/GameModsBR/PowerNukkit/pull/187/files/b8e4c2165d03c63632aee701f4de86d79baa2671#diff-049c0766b545977a6b865fa3409b25c8R1555

Reply:
done :D